build-conventions/src/main/kotlin/convention.control.gradle.kts

```diff
@@ -0,0 +1,69 @@
+import org.jetbrains.kotlin.gradle.dsl.KotlinJsProjectExtension
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmProjectExtension
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+import org.jetbrains.kotlin.konan.target.Family
+import org.jetbrains.kotlin.konan.target.HostManager
+import util.buildHost
+
+plugins {
+  id("convention.common")
+}
+
+pluginManager.withPlugin("org.jetbrains.kotlin.multiplatform") {
+  extensions.getByType(KotlinMultiplatformExtension::class.java).targets.let(::control)
+}
+pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+  objects.namedDomainObjectList(KotlinTarget::class.java).apply {
+    add(extensions.getByType(KotlinJvmProjectExtension::class.java).target)
+  }.let(::control)
+}
+pluginManager.withPlugin("org.jetbrains.kotlin.js") {
+  objects.namedDomainObjectList(KotlinTarget::class.java).apply {
+    add(extensions.getByType(KotlinJsProjectExtension::class.java).js())
```

Comment:
I know, not happy with it either, but this is the best I could come up with to avoid publishing cross-compiled artefacts twice when running on gh matrix. The whole purpose of this plugin is to make sure that when running on ci only the targets that are native to the given os are built and published (e.g. no linux builds on macos host)